Net/SMTP.php

```diff
@@ -1112,17 +1124,51 @@ protected function authGSSAPI($uid, $pwd, $authz = '')
     public function authXOAuth2($uid, $token, $authz, $conn)
```

Comment:
All auth* methods are protected, so this one should too, I suppose.

Reply:
I did not touch on previous methods already in place, I can correct it to move it into protected, no clue if it is already used outside of this library, I don't think so

Reply:
I corrected it in favor of a `protected` method
@schengawegga let me know your preferences

Reply:
My preference is like @alecpl, to protect all auth* methods, to be straight on this kind of methods.
But, i as you said @EdouardVanbelle, if anybody uses this method from outside, it will break backwards compatibility, and have to be released on a new major version. I also do not think, that anybody uses this from outside, but it is possible.

@alecpl @jparise what do you think about that? can we do a bugfix to change `authXOAuth2` method to protected without releasing this under a new major version?

Reply:
While waiting any feedback I kept the original signature and added a FIXME in comment